pkg/logs/tail_test.go

```diff
@@ -0,0 +1,323 @@
+			{SequenceNum: 7, RequestStart: 1601645172164667, Stream: "stdout", RequestID: "44a1eedd-5831-49fe-b094-7435908ba1fb", Message: "7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7"},
+			{SequenceNum: 8, RequestStart: 1601645172164667, Stream: "stdout", RequestID: "44a1eedd-5831-49fe-b094-7435908ba1fb", Message: "8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8 8"},
+			{SequenceNum: 9, RequestStart: 1601645172164667, Stream: "stderr", RequestID: "44a1eedd-5831-49fe-b094-7435908ba1fb", Message: "9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9 9"},
+			{SequenceNum: 10, RequestStart: 1601645172164667, Stream: "stdout", RequestID: "44a1eedd-5831-49fe-b094-7435908ba1fb", Message: "10 10 10 10 10 10 10 10 10 10 10 10 10 10"},
+		},
+	}
+
+	if diff := cmp.Diff(want, got); diff != "" {
+		t.Errorf("parseResponseData mismatch (-want +got):\n%s", diff)
+	}
+}
+
+func TestFilterStream(t *testing.T) {
+	for i, test := range []struct {
+		stream string
+		logs   []Log
+		explen int
+	}{
+		{
+			stream: "stdout",
+			logs: []Log{
+				{Stream: "stdout"},
+				{Stream: "stdout"},
+				{Stream: "stderr"},
+				{Stream: "stdout"},
+				{Stream: "stdout"},
+				{Stream: "stderr"},
+				{Stream: "stdout"},
+				{Stream: "stderr"},
+			},
+			explen: 5,
+		},
+		{
+			stream: "stderr",
+			logs: []Log{
+				{Stream: "stdout"},
+				{Stream: "stdout"},
+				{Stream: "stderr"},
+				{Stream: "stdout"},
+				{Stream: "stdout"},
+				{Stream: "stderr"},
+				{Stream: "stdout"},
+				{Stream: "stderr"},
+			},
+			explen: 3,
+		},
+		{
+			logs: []Log{
+				{Stream: "stdout"},
+				{Stream: "stdout"},
+				{Stream: "stderr"},
+				{Stream: "stderr"},
+				{Stream: "stdout"},
+				{Stream: "stderr"},
+				{Stream: "stdout"},
+				{Stream: "stderr"},
+			},
+			explen: 8,
+		},
+	} {
+		out := filterStream(test.stream, test.logs)
+		if len(out) != test.explen {
+			t.Errorf("#%d: exp: %d != got: %d", i, test.explen, len(out))
+		}
+	}
+}
+
+// TestGetLinks tests both getLinks() and getTimeFromLink()
+func TestGetLinks(t *testing.T) {
+	rawNexPrev := `</service/sid/log_stream/managed/instance_output%3Ffrom=1601412640>; rel="next", </service/sid/log_stream/managed/instance_output%3Ffrom=1601412620>; rel="prev"`
+	head := make(http.Header)
+	head.Set("Link", rawNexPrev)
+
+	prev, next := getLinks(head)
+	prevexp := "/service/sid/log_stream/managed/instance_output%3Ffrom=1601412620"
+	if prev != prevexp {
+		t.Errorf("prev header exp: %s != got: %s", prevexp, prev)
+	}
+
+	nextexp := "/service/sid/log_stream/managed/instance_output%3Ffrom=1601412640"
+	if next != nextexp {
+		t.Errorf("next header exp: %s != got: %s", nextexp, next)
+	}
+
+	pTime, err := getTimeFromLink(prev)
+	if err != nil {
+		t.Fatalf("unexpected error parsing prev link: %s", err)
+	}
+	exp := int64(1601412620)
+	if pTime != exp {
+		t.Errorf("prev time exp: %v != got: %v", exp, pTime)
+	}
```

Comment:
I guess for completeness we should also validate a `nTime` (i.e. the time is extracted from the 'next' link).